src/operator/nn/concat.cc

```diff
@@ -149,9 +149,10 @@ bool ConcatType(const nnvm::NodeAttrs& attrs,
     if (dtype == -1) {
       dtype = i;
     } else {
-      CHECK(i == dtype ||
-          i == -1) <<
-          "Non-uniform data type in Concat";
+      CHECK( i == dtype || i == -1) 
```

Comment:
Thanks for the fix! Two suggestions: 
- this function is used for both Concat and dstack. Can we extend the error msg to `Non-uniform data type in Concat / dstack.... ` ?
- can we extend the last sentenve and show the index of the data type : `..... got data type x at position y`?

Reply:
Do you want to split the msg explicitly of "Concat" and "dstack" based on the op name. Or just  use the sentence "Non-uniform data type in Concat / dstack....", and no need to state if it's Concat or dstack ? 

Yes, the second one can be done.

Reply:
Split the msg sounds more clear. Maybe we can template the function with the op name

Reply:
Add the op info in msg. Marking as resolved. Feel free to re-open.